core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -62,73 +41,91 @@
 import static org.apache.iceberg.TableProperties.SNAPSHOT_ID_INHERITANCE_ENABLED_DEFAULT;
 
 abstract class MergingSnapshotProducer<ThisT> extends SnapshotProducer<ThisT> {
-  private static final Logger LOG = LoggerFactory.getLogger(MergingSnapshotProducer.class);
+  private final String tableName;
+  private final TableOperations ops;
+  private final PartitionSpec spec;
+  private final SnapshotSummary.Builder summaryBuilder = SnapshotSummary.builder();
+  private final ManifestMergeManager<DataFile> mergeManager;
+  private final ManifestFilterManager<DataFile> filterManager;
+  private final boolean snapshotIdInheritanceEnabled;
 
-  private static final Joiner COMMA = Joiner.on(",");
+  private class DataFileFilterManager extends ManifestFilterManager<DataFile> {
```

Comment:
nit: Would it make sense to move private classes to the bottom of the file so that people will focus on the main logic?

Reply:
I do that in #1105.